### PR TITLE
Add workflow to verify binary installation

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
I have observed some issues within other plugins of issues being only caught at run time. This is because several things can go wrong during the build process, which may not be caught in a dev setup. This adds a workflow to verify that building and installing into OSD works on every PR.

Related issues:
https://github.com/opensearch-project/security-dashboards-plugin/issues/1709
https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/875
https://github.com/opensearch-project/dashboards-observability/pull/309
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5952

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by derek-ho** [comment]
Seems like there is an issue with a recent dependency OSD core added. Created a follow up issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5953. @ryanbogan can you follow up here and help take this PR to completion? Thanks!

File: .github/workflows/verify-binary-installation.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          opensearch-version: ${{ env.OPENSEARCH_VERSION }}
+          security-enabled: false
+
+      - name: Run Dashboard
+        id: setup-dashboards
+        uses: derek-ho/setup-opensearch-dashboards@v2
+        with:
+          plugin_name: dashboards-maps
+          built_plugin_name: mapsDashboards
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply .github/workflows/verify-binary-installation.yml:42]: Thanks @derek-ho, the plugin name is `customImportMapDashboards`. https://github.com/opensearch-project/dashboards-maps/blob/main/opensearch_dashboards.json#L2

[the remark this answers is not in the document]
Comment (derek-ho) [reply .github/workflows/verify-binary-installation.yml]: Thanks updated here: [af0bff2](https://github.com/opensearch-project/dashboards-maps/pull/591/commits/af0bff23ecc3e1855df41b1d89224be2f6f9f33e), but need to re-run it to confirm once the bootstrapping issue is resolved. Let me know if there is any other action item for me

**Review by junqiu-lei — APPROVED** [review]

File: .github/workflows/verify-binary-installation.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Start the binary
+        run: | 
+          nohup ./bin/opensearch-dashboards &
+        working-directory: ${{ steps.setup-dashboards.outputs.dashboards-binary-directory }}
+        shell: bash
+
+      - name: Health check 
+        run: |
+          timeout 300 bash -c 'while [[ "$(curl http://localhost:5601/api/status | jq -r '.status.overall.state')" != "green" ]]; do sleep 5; done'
```
Comment (martin-gaievski) [inline .github/workflows/verify-binary-installation.yml:54]: is 300 seconds enough timeout based on previous experience?

Comment (derek-ho) [reply .github/workflows/verify-binary-installation.yml:54]: Yea - this is the OSD binary/not a dev setup, so we don't need to wait for the bundles to compile

Comment (derek-ho) [reply .github/workflows/verify-binary-installation.yml:54]: This run the health check was successful almost right away: https://github.com/opensearch-project/dashboards-maps/actions/runs/8912664392/job/24476554624?pr=591

**Review by martin-gaievski — APPROVED** [review]

**Pull request merged**